src/runtime_src/core/common/device_factory.h

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2023 Advanced Micro Devices, Inc. All rights reserved.
+#ifndef _XCL_DEV_FACTORY_H_
+#define _XCL_DEV_FACTORY_H_
+
+#include "device.h"
+#include <vector>
+
+namespace xrt_core {
+
+// One function on FPGA or AIE device
+class device_factory
+{
+public:
+  bool m_is_mgmt = false;
+  bool m_is_ready = true; 
+  device_factory(bool isuser) {  
+      m_is_mgmt = !isuser;  
+  }
+  
+  virtual ~device_factory() {}
+ 
+  // Hand out a "device" instance that is specific to this type of device.
+  // Caller will use this device to access device specific implementation of ishim.
+  virtual std::shared_ptr<device>
+  create_device(device::handle_type handle, device::id_type id) const = 0;
+
+  // Hand out an opaque "shim" handle that is specific to this type of device.
+  // On legacy Alveo device, this handle can be used to lookup a device instance and
+  // make xcl HAL API calls.
+  // On new platforms, this handle can only be used to look up a device. HAL API calls
+  // through it are not supported any more.
+  virtual device::handle_type
+  create_shim(device::id_type id) const = 0;
+
+  //get BDF 
```

Comment:
Be consistent in commenting.  Yes, this one is quit obvious, but expand on what is returned, as it definitely isn't just BDF since it is a tuple of 4.  It's more like DBDF, but do explain.
```suggestion
  // Get BDF 
```

Reply:
done